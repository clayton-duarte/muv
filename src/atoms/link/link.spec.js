import React from 'react';

import Link from '.';

it('should render a button', () => {
  const wrapper = shallow(
    <Link onClick={() => true} />,
  ).dive();
  expect(wrapper).toMatchSnapshot();
});


it('should render a anchor', () => {
  const wrapper = shallow(
    <Link href="#" />,
  ).dive();
  expect(wrapper).toMatchSnapshot();
});

it('should change its style on warn', () => {
  const wrapper = shallow(
    <Link warn href="#" />,
  ).dive();
  expect(wrapper).toMatchSnapshot();
});

it('should change its style on success', () => {
  const wrapper = shallow(
    <Link success href="#" />,
  ).dive();
  expect(wrapper).toMatchSnapshot();
});

it('should change its style on secondary', () => {
  const wrapper = shallow(
    <Link secondary href="#" />,
  ).dive();
  expect(wrapper).toMatchSnapshot();
});
