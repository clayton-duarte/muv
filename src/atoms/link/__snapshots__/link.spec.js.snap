// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change its style on secondary 1`] = `
<Link
  disabled={false}
  href="#"
  noLine={false}
  onClick={null}
  secondary={true}
  strong={false}
>
  <styled.a
    disabled={false}
    href="#"
    noLine={false}
    onClick={null}
    secondary={true}
    strong={false}
    theme={
      Object {
        "color": Object {
          "black": "#222222",
          "darkgray": "#515151",
          "error": "#dd3333",
          "gray": "#a9abae",
          "lightgray": "#e5e5e5",
          "overlay": "rgba(0, 0, 0, .15)",
          "primary": "#0081c8",
          "secondary": "#00245f",
          "success": "#33bb33",
          "warn": "#dd3333",
          "white": "#fefefe",
        },
        "font": Object {
          "color": "#222222",
          "family": "\\"Ubuntu\\", sans-serif",
          "size": "16px",
          "spacing": 0,
        },
        "shape": Object {
          "border": ".125rem solid rgba(0, 0, 0, .15)",
          "float": "0 .5rem .5rem 0 rgba(0, 0, 0, .15)",
          "margin": ".25rem .125rem",
          "padding": ".75rem .875rem",
          "radius": ".25rem",
          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
          "size": "16px",
        },
        "transition": Object {
          "time": ".3s ease",
        },
      }
    }
  >
    <a
      className="sc-bdVaJa jGYEHc"
      disabled={false}
      href="#"
      onClick={null}
    >
      default
    </a>
  </styled.a>
</Link>
`;

exports[`should change its style on success 1`] = `
<Link
  disabled={false}
  href="#"
  noLine={false}
  onClick={null}
  strong={false}
  success={true}
>
  <styled.a
    disabled={false}
    href="#"
    noLine={false}
    onClick={null}
    strong={false}
    success={true}
    theme={
      Object {
        "color": Object {
          "black": "#222222",
          "darkgray": "#515151",
          "error": "#dd3333",
          "gray": "#a9abae",
          "lightgray": "#e5e5e5",
          "overlay": "rgba(0, 0, 0, .15)",
          "primary": "#0081c8",
          "secondary": "#00245f",
          "success": "#33bb33",
          "warn": "#dd3333",
          "white": "#fefefe",
        },
        "font": Object {
          "color": "#222222",
          "family": "\\"Ubuntu\\", sans-serif",
          "size": "16px",
          "spacing": 0,
        },
        "shape": Object {
          "border": ".125rem solid rgba(0, 0, 0, .15)",
          "float": "0 .5rem .5rem 0 rgba(0, 0, 0, .15)",
          "margin": ".25rem .125rem",
          "padding": ".75rem .875rem",
          "radius": ".25rem",
          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
          "size": "16px",
        },
        "transition": Object {
          "time": ".3s ease",
        },
      }
    }
  >
    <a
      className="sc-bdVaJa cALpdh"
      disabled={false}
      href="#"
      onClick={null}
    >
      default
    </a>
  </styled.a>
</Link>
`;

exports[`should change its style on warn 1`] = `
<Link
  disabled={false}
  href="#"
  noLine={false}
  onClick={null}
  strong={false}
  warn={true}
>
  <styled.a
    disabled={false}
    href="#"
    noLine={false}
    onClick={null}
    strong={false}
    theme={
      Object {
        "color": Object {
          "black": "#222222",
          "darkgray": "#515151",
          "error": "#dd3333",
          "gray": "#a9abae",
          "lightgray": "#e5e5e5",
          "overlay": "rgba(0, 0, 0, .15)",
          "primary": "#0081c8",
          "secondary": "#00245f",
          "success": "#33bb33",
          "warn": "#dd3333",
          "white": "#fefefe",
        },
        "font": Object {
          "color": "#222222",
          "family": "\\"Ubuntu\\", sans-serif",
          "size": "16px",
          "spacing": 0,
        },
        "shape": Object {
          "border": ".125rem solid rgba(0, 0, 0, .15)",
          "float": "0 .5rem .5rem 0 rgba(0, 0, 0, .15)",
          "margin": ".25rem .125rem",
          "padding": ".75rem .875rem",
          "radius": ".25rem",
          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
          "size": "16px",
        },
        "transition": Object {
          "time": ".3s ease",
        },
      }
    }
    warn={true}
  >
    <a
      className="sc-bdVaJa hCHyHr"
      disabled={false}
      href="#"
      onClick={null}
    >
      default
    </a>
  </styled.a>
</Link>
`;

exports[`should render a anchor 1`] = `
<Link
  disabled={false}
  href="#"
  noLine={false}
  onClick={null}
  strong={false}
>
  <styled.a
    disabled={false}
    href="#"
    noLine={false}
    onClick={null}
    strong={false}
    theme={
      Object {
        "color": Object {
          "black": "#222222",
          "darkgray": "#515151",
          "error": "#dd3333",
          "gray": "#a9abae",
          "lightgray": "#e5e5e5",
          "overlay": "rgba(0, 0, 0, .15)",
          "primary": "#0081c8",
          "secondary": "#00245f",
          "success": "#33bb33",
          "warn": "#dd3333",
          "white": "#fefefe",
        },
        "font": Object {
          "color": "#222222",
          "family": "\\"Ubuntu\\", sans-serif",
          "size": "16px",
          "spacing": 0,
        },
        "shape": Object {
          "border": ".125rem solid rgba(0, 0, 0, .15)",
          "float": "0 .5rem .5rem 0 rgba(0, 0, 0, .15)",
          "margin": ".25rem .125rem",
          "padding": ".75rem .875rem",
          "radius": ".25rem",
          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
          "size": "16px",
        },
        "transition": Object {
          "time": ".3s ease",
        },
      }
    }
  >
    <a
      className="sc-bdVaJa lhHpOn"
      disabled={false}
      href="#"
      onClick={null}
    >
      default
    </a>
  </styled.a>
</Link>
`;

exports[`should render a button 1`] = `
<Link
  disabled={false}
  href={null}
  noLine={false}
  onClick={[Function]}
  strong={false}
>
  <styled.button
    disabled={false}
    href={null}
    noLine={false}
    onClick={[Function]}
    strong={false}
    theme={
      Object {
        "color": Object {
          "black": "#222222",
          "darkgray": "#515151",
          "error": "#dd3333",
          "gray": "#a9abae",
          "lightgray": "#e5e5e5",
          "overlay": "rgba(0, 0, 0, .15)",
          "primary": "#0081c8",
          "secondary": "#00245f",
          "success": "#33bb33",
          "warn": "#dd3333",
          "white": "#fefefe",
        },
        "font": Object {
          "color": "#222222",
          "family": "\\"Ubuntu\\", sans-serif",
          "size": "16px",
          "spacing": 0,
        },
        "shape": Object {
          "border": ".125rem solid rgba(0, 0, 0, .15)",
          "float": "0 .5rem .5rem 0 rgba(0, 0, 0, .15)",
          "margin": ".25rem .125rem",
          "padding": ".75rem .875rem",
          "radius": ".25rem",
          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
          "size": "16px",
        },
        "transition": Object {
          "time": ".3s ease",
        },
      }
    }
  >
    <button
      className="sc-bwzfXH mRfpw"
      disabled={false}
      href={null}
      onClick={[Function]}
    >
      default
    </button>
  </styled.button>
</Link>
`;
